### PR TITLE
General Clean-ups of Documentation and Public API ahead of first release

### PR DESCRIPTION
**Pull request by Carter12s**

### CODE REVIEW

File: roslibrust/src/comm.rs [hunk 1]
```diff
@@ -76,7 +80,7 @@ impl FromStr for Ops {
 
 /// Describes the low level comm capabilities of talking to a rosbridge server
 #[async_trait]
-pub trait RosBridgeComm {
+pub(crate) trait RosBridgeComm {
```
Comment (ssnover) [inline roslibrust/src/comm.rs:83]: Is the intention to not allow external code to impl this trait?

**Review by ssnover — APPROVED** [review]
All of the comments look good. Unfortunately, many of them are on codegen which I've just finished rewriting :smile:

**Comment by ssnover** [comment]
One thing I'd advocate for (maybe prior to publishing on crates.io) is to recreate the ROS tutorial in a mdbook and then host it with Jekyll from a GitHub repo. This will make it easier to show how to get started.

File: roslibrust/src/comm.rs (hunk 1 above)
Comment (Carter12s) [reply roslibrust/src/comm.rs:83]: It is "for now" I don't think anyone should be impl-ing until we have a better plan going forward.

**Comment by Carter12s** [comment]
> One thing I'd advocate for (maybe prior to publishing on crates.io) is to recreate the ROS tutorial in a mdbook and then host it with Jekyll from a GitHub repo. This will make it easier to show how to get started.

Filed https://github.com/Carter12s/roslibrust/issues/22

**Pull request merged**